src/Core/src/Platform/Android/MauiAppCompatActivity.cs

```diff
@@ -44,5 +44,77 @@ public override bool DispatchTouchEvent(MotionEvent? e)
 
 			return handled || implHandled;
 		}
+
+		// Internal methods for key event forwarding from modal dialogs
+		// These call the user's overrides without calling base to avoid side effects
+		[System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
+		internal bool InternalOnKeyDown(Keycode keyCode, KeyEvent? e)
+		{
+			if (_processingKeyEvent)
+				return false;
+
+			_processingKeyEvent = true;
+			try
+			{
+				return OnKeyDown(keyCode, e);
```

Comment:
I was thinking we'd then override OnKeyDown

And inside the override we don't call thse base if the flag is set

If you like that idea then apply it everywhere